qiskit/transpiler/passes/scheduling/alignments/reschedule.py

```diff
@@ -191,7 +200,7 @@ def run(self, dag: DAGCircuit):
         2. If the start time of the node violates the alignment constraints,
             the scheduler increases the start time until it satisfies the constraint.
         3. Check overlap with successor nodes. If any overlap occurs, the rescheduler
-            recursively pushs the successor nodes backward towards the end of the wire.
+            pushes the successor nodes backward towards the end of the wire.
             Note that shifted location doesn't need to satisfy the constraints,
             thus it will be a minimum delay to resolve the overlap with the ancestor node.
```

Comment:
Can you update the docstring here to call out that we'll unconditionally call `_push_node_back` to catch either misalignments or cases where downstream nodes need to be pushed back to avoid overlap?

Reply:
I made a pass over the whole docstring, hopefully making it a bit more clear. Please see [b1f9c7c](https://github.com/Qiskit/qiskit-terra/pull/10077/commits/b1f9c7c7c63943265e86ca37ffebca1616dde9bb).